x-pack/plugins/security_solution/public/common/components/exceptions/builder/and_badge.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiFlexItem } from '@elastic/eui';
+import styled from 'styled-components';
+
+import { AndOrBadge } from '../../and_or_badge';
+
+const MyInvisibleAndBadge = styled(EuiFlexItem)`
+  visibility: hidden;
+`;
+
+const MyFirstRowContainer = styled(EuiFlexItem)`
+  padding-top: 20px;
+`;
+
+interface BuilderAndBadgeProps {
+  entriesLength: number;
+  exceptionItemIndex: number;
+  andLogicIncluded: boolean;
+}
+
+export const BuilderAndBadgeComponent = React.memo<BuilderAndBadgeProps>(
+  ({ entriesLength, exceptionItemIndex, andLogicIncluded }) => {
+    const badge = <AndOrBadge includeAntennas type="and" />;
+
+    if (andLogicIncluded && entriesLength > 1 && exceptionItemIndex === 0) {
+      return (
+        <MyFirstRowContainer grow={false} data-test-subj="exceptionItemEntryFirstRowAndBadge">
+          {badge}
+        </MyFirstRowContainer>
+      );
+    } else if (andLogicIncluded && entriesLength <= 1) {
+      return (
+        <MyInvisibleAndBadge grow={false} data-test-subj="exceptionItemEntryInvisibleAndBadge">
+          {badge}
+        </MyInvisibleAndBadge>
+      );
+    } else if (andLogicIncluded && entriesLength > 1) {
+      return (
+        <EuiFlexItem grow={false} data-test-subj="exceptionItemEntryAndBadge">
+          {badge}
+        </EuiFlexItem>
+      );
+    } else {
+      return <></>;
```

Comment:
Would returning `null` also work here?

Reply:
This is just a nit, but it's a little weird that `andLogicIncluded` is passed in, but nothing happens if it's `false`. I think I'd rather see something like `{andLogicIncluded && BuilderAndBadgeComponent(...)}` without passing the boolean flag in.

Reply:
@madirey that's a good point. I've updated to remove it as you suggested. @peluja1012 that also then addresses your concern as the default state returns a non empty component.